lib/permutations.rb

```diff
@@ -1,4 +1,15 @@
 
 def permutations?(string1, string2)
-  raise NotImplementedError, "permutations? not implemented"
+  return false if string1.length != string2.length 
+  hash_table = {}
+  (string1.chars).each do |s|
+    hash_table[s] = 1 
```

Comment:
This checks that they have the same letters, but not the same **number** of letters on each side.  Instead you should keep a count rather than just 1 if the letter is taken.